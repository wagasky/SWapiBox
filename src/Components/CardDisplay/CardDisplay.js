import React from 'react';
import People from '../People/People'
import Vehicles from '../Vehicles/Vehicles'
import Planets from '../Planets/Planets'
import Favorites from '../Favorites/Favorites'
import IntroText from '../IntroText/IntroText'
import { Switch, Route } from 'react-router-dom';

const CardDisplay = ({ people, planets, vehicles, favorites, film, handleFavorite }) => {

return (
  <div>
  <p>I'm a CardDisplay!</p>
    <Switch>
      <Route exact path='/people' component={ () => <People people={people} handleFavorite={handleFavorite} /> } />
      <Route exact path='/vehicles' component={ () => <Vehicles vehicles={vehicles} /> }/>
      <Route exact path='/planets' component={ () => <Planets planets={planets} /> } />
      <Route exact path='/favorites' component={ () => <Favorites favorites={favorites} /> }/>
    </Switch>
    <IntroText film={film}/>
  </div>
  )

}

export default CardDisplay;



