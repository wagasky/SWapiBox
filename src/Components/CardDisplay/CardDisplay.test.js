/* eslint-disable */ 

import React from 'react';
import ReactDOM from 'react-dom';
import CardDisplay from './CardDisplay';
import { shallow } from 'enzyme';

it.skip('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<CardDisplay />, div);
  ReactDOM.unmountComponentAtNode(div);
});