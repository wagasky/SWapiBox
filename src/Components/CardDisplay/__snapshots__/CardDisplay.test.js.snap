// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDisplay should match snapshot 1`] = `
<div>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/people"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/vehicles"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/planets"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/favorites"
    />
  </Switch>
  <IntroText
    film={
      Array [
        Object {
          "number": "IV",
          "releaseYear": "1977-05-25",
          "text": "It is a period of civil war.
      Rebel spaceships, striking
      from a hidden base, have won
      their first victory against
      the evil Galactic Empire.
      
      During the battle, Rebel
      spies managed to steal secret
      plans to the Empire's
      ultimate weapon, the DEATH
      STAR, an armored space
      station with enough power
      to destroy an entire planet.
      
      Pursued by the Empire's
      sinister agents, Princess
      Leia races home aboard her
      starship, custodian of the
      stolen plans that can save her
      people and restore
      freedom to the galaxy....",
          "title": "A New Hope",
        },
      ]
    }
  />
</div>
`;
